report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java

```diff
@@ -17,73 +16,262 @@
+                new LocalDate(2023, 01, 1),
+                new LocalDate(2023, 01, 31));
+
+        List<Map<String, Object>> result = getReportResult(query);
+
+        // Assert
+        assertEquals(result.get(0).get("Title"), "Number of people tested in the month disaggregated by entry point");
+        assertEquals(result.get(0).get("Index"), 0);
+        assertEquals(result.get(0).get("Emergency"), 0);
+        assertEquals(result.get(0).get("CPN / ANC"), 0);
+        assertEquals(result.get(0).get("SA/Maternité / Delivery room/Postpartumn"), 0);
+        assertEquals(result.get(0).get("VCT"), 0);
+        assertEquals(result.get(0).get("Hospitalization"), 0);
+        assertEquals(result.get(0).get("Pediatrics"), 0);
+        assertEquals(result.get(0).get("Other testing at TB Unit"), 0);
+        assertEquals(result.get(0).get("Blood Bank"), 0);
+        assertEquals(result.get(0).get("Screening Campaign"), 0);
+        assertEquals(result.get(0).get("Other entry Points"), 0);
+
+    }
+
+    @Test
+    public void shouldNotCountHivPatientWhenThereIsNoResult() throws Exception {
+        // register a new patient
+        int patientId = testDataGenerator.registration.createPatient(
+                GenderEnum.MALE,
+                new LocalDate(2000, 9, 1),
+                "John",
+                "Doe");
+
+        // start a new visit
+        int encounterId = testDataGenerator.startVisit(
+                patientId,
+                new LocalDate(2022, 11, 1),
+                VisitTypeEnum.VISIT_TYPE_OPD);
+
+        // enrol patient into hiv program
+        testDataGenerator.program.enrollPatientIntoHIVProgram(
+                patientId,
+                new LocalDate(2022, 11, 3),
+                ConceptEnum.WHO_STAGE_1,
+                TherapeuticLineEnum.FIRST_LINE,
+                new LocalDate(2022, 11, 3));
+
+        // set hiv test date
+        testDataGenerator.hivTestingAndCounsellingForm.setHTCHivTestDate(
+                patientId,
+                new LocalDateTime(2022, 11, 5, 8, 0),
+                new LocalDate(2022, 11, 5),
+                encounterId);
+
+        // set entrypoints
         testDataGenerator.hivTestingAndCounsellingForm.recordTestingEntryPointAndModality(
-            patientId,
-			new LocalDateTime(2022, 11, 11, 8, 0),
-            ConceptEnum.VCT,
-            encounterId
-        );
+                patientId,
+                new LocalDateTime(2022, 11, 11, 8, 0),
+                ConceptEnum.VCT,
+                encounterId);
 
+        // execute
         String query = readReportQuery(ReportEnum.CDV_TS_REPORT,
-         "indicator1_CDV_TS_number_of_people_tested_in_the_month_disaggregation_entry_point.sql",
-          new LocalDate(2022, 11, 1),
-          new LocalDate(2022, 11, 30)
-          );
+                "indicator1_CDV_TS_number_of_people_tested_in_the_month_disaggregation_entry_point.sql",
+                new LocalDate(2023, 01, 1),
```

Comment:
@tresorluz for this test, the HIV test date needs to be within the reporting period (i.e. we need to use nov 2022 here)